[@analogjs/vitest-angular] latest beta has incorrect dependency version requirement on @angular-devkit/architect

### Please provide the environment you discovered this bug in.

I've tested on both mac and windows. With and without `--force`. Including latest release `1.4.0-beta.7`

### Which area/package is the issue in?

vitest-angular

### Description

I appreciate the work to separate the vitest-angular logic into its own package. This should make my team's incremental adoption of the vite plugin much easier.

The issue I'm having is that my application doesn't currently have any dependency on `@angular-devkit/architect`. When I install the latest, it's still blocked from installing the `@angularjs/vitest-angular` plugin because the versionings don't match.

The version naming convention for the `@angular-devkit/architect` package does not adhere to the same angular versioning conventions.

<img width="780" alt="image" src="https://github.com/analogjs/analog/assets/130445/0a488c3e-e479-4e1f-ba14-c358b9d6bb2a">


### Please provide the exception or error you saw

```true
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @angular-devkit/architect@^15.0.0 || ^16.0.0 || ^17.0.0 || ^18.0.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```


### Other information

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

--- a/packages/vitest-angular/package.json
+++ b/packages/vitest-angular/package.json
@@ -29,7 +29,7 @@
   },
   "peerDependencies": {
     "@analogjs/vite-plugin-angular": "^1.3.0",
-    "@angular-devkit/architect": "^15.0.0 || ^16.0.0 || ^17.0.0 || ^18.0.0",
+    "@angular-devkit/architect": "0.1500.0-0.1599.999 || 0.1600.0-0.1699.999 || 0.1700.0-0.1799.999 || 0.1800.0-0.1899.999",
     "vitest": "^1.3.1"
   },
   "ng-update": {

--- a/packages/vitest-angular/package.json
+++ b/packages/vitest-angular/package.json
@@ -29,7 +29,7 @@
   },
   "peerDependencies": {
     "@analogjs/vite-plugin-angular": "^1.3.0",
-    "@angular-devkit/architect": "0.1500.0-0.1599.999 || 0.1600.0-0.1699.999 || 0.1700.0-0.1799.999 || 0.1800.0-0.1899.999",
+    "@angular-devkit/architect": "^0.1500.0 || ^0.1600.0 || ^0.1700.0 || ^0.1800.0",
     "vitest": "^1.3.1"
   },
   "ng-update": {

